app/gui/controller/double-representation/src/node.rs

```diff
@@ -357,13 +465,13 @@ impl MainLine {
         }
     }
 
-    /// Set the pattern (left side of assignment) for node. If it is an Expression node, the
-    /// assignment infix will be introduced.
-    pub fn set_pattern(&mut self, pattern: Ast) {
+    /// See [`NodeInfo::set_pattern`]. Returns the AST info if it was updated.
+    fn set_pattern(&mut self, pattern: Ast) -> Option<NodeAstInfo> {
```

Comment:
Is there a scenario where the NodeAstInfo could change after pattern change?

Reply:
Not, good catch. I was confused by implementation.